src/sql/azdata.d.ts

```diff
@@ -2555,37 +2555,37 @@ declare module 'azdata' {
 	 * Supports defining a model that can be instantiated as a view in the UI
 	 */
 	export interface ModelBuilder {
-		navContainer(): ContainerBuilder<NavContainer, any, any>;
+		navContainer(): ContainerBuilder<NavContainer, any, any, ComponentProperties>;
 		divContainer(): DivBuilder;
 		flexContainer(): FlexBuilder;
 		splitViewContainer(): SplitViewBuilder;
-		dom(): ComponentBuilder<DomComponent>;
+		dom(): ComponentBuilder<DomComponent, DomProperties>;
 		/**
 		 * @deprecated please use radioCardGroup component.
 		 */
-		card(): ComponentBuilder<CardComponent>;
-		inputBox(): ComponentBuilder<InputBoxComponent>;
-		checkBox(): ComponentBuilder<CheckBoxComponent>;
-		radioButton(): ComponentBuilder<RadioButtonComponent>;
-		webView(): ComponentBuilder<WebViewComponent>;
-		editor(): ComponentBuilder<EditorComponent>;
-		diffeditor(): ComponentBuilder<DiffEditorComponent>;
-		text(): ComponentBuilder<TextComponent>;
-		image(): ComponentBuilder<ImageComponent>;
-		button(): ComponentBuilder<ButtonComponent>;
-		dropDown(): ComponentBuilder<DropDownComponent>;
-		tree<T>(): ComponentBuilder<TreeComponent<T>>;
-		listBox(): ComponentBuilder<ListBoxComponent>;
-		table(): ComponentBuilder<TableComponent>;
-		declarativeTable(): ComponentBuilder<DeclarativeTableComponent>;
-		dashboardWidget(widgetId: string): ComponentBuilder<DashboardWidgetComponent>;
-		dashboardWebview(webviewId: string): ComponentBuilder<DashboardWebviewComponent>;
+		card(): ComponentBuilder<CardComponent, CardProperties>;
+		inputBox(): ComponentBuilder<InputBoxComponent, InputBoxProperties>;
+		checkBox(): ComponentBuilder<CheckBoxComponent, CheckBoxProperties>;
+		radioButton(): ComponentBuilder<RadioButtonComponent, RadioButtonProperties>;
+		webView(): ComponentBuilder<WebViewComponent, WebViewProperties>;
+		editor(): ComponentBuilder<EditorComponent, EditorProperties>;
+		diffeditor(): ComponentBuilder<DiffEditorComponent, DiffEditorComponent>;
```

Comment:
Should this be `DiffEditorProperties`?

Reply:
Unfortunately there is no diffEdtiorProperties. I think the authors of that component made a few mistakes in designing it, but I'm not trying to address those now.